### ISSUE TITLE
Some Go tests failed with new Go 1.6

### ISSUE THREAD

**Issue by jplesnik** [swig/swig#619]
I try to build Swig on system with new Go 1.6 and some tests failed with following errors:

```
checking Examples/go/director
panic: runtime error: cgo result has Go pointer

goroutine 1 [running]:
panic(0x4e4120, 0xc82000a410)
	/usr/lib/golang/src/runtime/panic.go:464 +0x3e6
/root/swig/Examples/go/director/example._cgoexpwrap_f85789ef08d9_Swig_DirectorFooBarAbstract_callback_Foo_example_1a396cf81700e959.func1(0xc820045bf8)
	??:0 +0x66
/root/swig/Examples/go/director
/example._cgoexpwrap_f85789ef08d9_Swig_DirectorFooBarAbstract_callback_Foo_example_1a396cf81700e959(0x1, 0xc82000a3f4, 0x6)
	??:0 +0x83
/root/swig/Examples/go/director/example._Cfunc__wrap_FooBarAbstract_FooBar_example_1a396cf81700e959(0x11aedc0, 0x0, 0x0)
	??:0 +0x44
/root/swig/Examples/go/director/example.SwigcptrFooBarAbstract.FooBar(0x11aedc0, 0x0, 0x0)
	/root/swig/Examples/go/director/gopath/src/example/example.go:186 +0x3f
/root/swig/Examples/go/director/example.(*_swig_DirectorFooBarAbstract).FooBar(0xc82000e260, 0x0, 0x0)
	<autogenerated>:23 +0x48
/root/swig/Examples/go/director/example.(*fooBarGo).FooBar(0xc82000a3d0, 0x0, 0x0)
	<autogenerated>:7 +0x5d
main.TestFooBarGo(0x0, 0x0)
	/root/swig/Examples/go/director/runme.go:26 +0x7d
main.main()
	/root/swig/Examples/go/director/runme.go:33 +0x208
../../Makefile:1995: recipe for target 'go_run' failed
make[3]: *** [go_run] Error 2
Makefile:11: recipe for target 'check' failed
make[2]: *** [check] Error 2
Makefile:250: recipe for target 'director.actionexample' failed
make[1]: *** [director.actionexample] Error 2
```

```
checking go testcase go_inout (with run test)
panic: runtime error: cgo argument has Go pointer to Go pointer

goroutine 1 [running]:
panic(0x51b440, 0xc82000a620)
	/usr/lib/golang/src/runtime/panic.go:464 +0x3e6
/root/swig/Examples/test-suite/go/go_inout.DoubleArray(0xc820043ef0)
	/root/swig/Examples/test-suite/go/gopath/src/go_inout/go_inout.go:278 +0x4a6
main.main()
	/root/swig/Examples/test-suite/go/go_inout_runme.go:37 +0x604
Makefile:49: recipe for target 'go_inout.cpptest' failed
make[1]: *** [go_inout.cpptest] Error 2
```

I was building Swig 3.0.8 and the version of Go

```
$ go version
go version go1.6 linux/amd64
```

It is probably related to this change https://golang.org/doc/go1.6#cgo

**Comment by wsfulton** [comment]
@ianlancetaylor can you add in Go 1.6 support?

**Comment by ianlancetaylor** [comment]
This is now fixed by 223c2a483563e5c9dd93067831a6a2af6252bcec.
